language/tools/mirai-dataflow-analysis/src/main.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use mirai_annotations::unrecoverable;
+use regex::Regex;
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use structopt::StructOpt;
+
+mod configuration;
+mod datalog;
+mod ddlog_output;
+mod souffle_output;
+mod util;
+
+use configuration::generate_config;
+use datalog::{decode_analysis_output, DatalogBackend, DatalogRelations};
+use ddlog_output::{parse_ddlog_output, run_ddlog_analysis};
+use souffle_output::{parse_souffle_output, run_souffle_analysis};
+use util::{get_child_output, NodeMap, TypeMap};
+
+// Run the Datalog analysis on the test file and
+// capture output
+fn run_mirai(config_path: &Path, crate_path: &Path, no_clean: bool) -> Result<(), String> {
+    if !no_clean {
+        // cargo clean
```

Comment:
The following code does not seem to be a `cargo clean` to me

Reply:
Good catch!